### PR TITLE
Add impl of map_put, map_has, map_get

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What

Add impl of map_put, map_has, map_get.

### Why

Need it for the SDK for user-defined type encoding as maps.

### Known limitations

[TODO or N/A]

### CODE REVIEW

**Comment by leighmcculloch** [comment]
@jayz22 Not sure if I'm doing this correctly, please take a look when you have a moment even if it is post-merge.

File: stellar-contract-env-host/src/host.rs [hunk 1]
```diff
@@ -597,11 +597,18 @@ impl CheckedEnv for Host {
     }
 
     fn map_put(&self, m: Object, k: RawVal, v: RawVal) -> Result<Object, HostError> {
-        todo!()
+        let k = self.associate_raw_val(k);
+        let v = self.associate_raw_val(v);
+        let mnew = self.visit_obj(m, |hm: &HostMap| Ok(hm.update(k, v)))?;
```
Comment (jayz22) [inline stellar-contract-env-host/src/host.rs:602]: Cool, I didn't see the `update` function and used `insert` instead. This looks much better.

**Comment by leighmcculloch** [comment]
@jayz22 I realize now that you implemented map already in #131. I missed it because I didn't see "map" in the title oops. In any case it looks like my implementation is almost identical to yours. Sorry for the conflict!

**Review by jayz22 — COMMENTED** [review]
Looks great! 
I've added most of the map and binary function in https://github.com/stellar/rs-stellar-contract-env/pull/131 . 
Feel free to copy them if needed. Will try to get that one merged soon.

File: stellar-contract-env-host/src/host.rs (hunk 1 above)
Comment (leighmcculloch) [reply stellar-contract-env-host/src/host.rs:602]: Ah interesting, I didn't see insert. It looks like they do almost the same thing. The API for these collection types are interesting and not completely intuitive to me. i.e. I find it confusing that 'update' returns a copy and doesn't actually update the existing map, whilst 'insert' does. 🤷🏻

Comment (jayz22) [reply stellar-contract-env-host/src/host.rs:602]: Haha my thoughts exactly.

Comment (jayz22) [reply stellar-contract-env-host/src/host.rs:602]: Yeah, `update` is just a `clone + insert`: https://docs.rs/im/latest/src/im/ord/map.rs.html#832-836

**Pull request merged**